draft-ietf-mls-protocol.md

```diff
@@ -1063,11 +1065,13 @@ it for any further processing.
 uint64 expiration;
 ~~~~~
 
-Note that as an extension, it is not required that any given ClientInitKey have
-an expiration time.  In particular, applications that rely on "last resort"
-ClientInitKeys to ensure continued reachability may choose to omit the
-expiration extension from these keys, or give them much longer lifetimes than
-other ClientInitKeys.
+Applications that rely on "last resort" ClientInitKeys MAY set the
+expiration to its maximum value even though this is NOT RECOMMENDED.
+It is RECOMMENDED to rotate last resort keys at a pace chosen by the
+application even though they can have much longer lifetimes than other
+ClientInitKeys.
```

Comment:
The expiration of ClientInitKeys is important for PCS, so setting to the maximum value shouldn't be encouraged. If you have keys that never expire and you're compromised, those keys can be used at any time in the future to "add" you to a group when they're actually giving the adversary access.

Last Resort keys are a defense against DoS, when somebody sends you a lot of messages in a short period of time. As such, any ClientInitKey can be a Last Resort key, regardless of how long it's valid for, because nobody is checking how many times a specific key gets used.

Last Resort keys are not a suitable defense against the user being offline for an arbitrary amount of time. Handling this securely probably requires more consideration.

We should recommend an application-specified maximum duration, and switch to a `(notBefore, notAfter)` system so that everyone know's the key's lifetime.